File: week06/20/Pr5.c
int maxNumberOfBalloons(char* text) {
    // 노가다스럽지만, 아이디어는 좋다
    int ans[26] = {0};

    for(int i=0;text[i]!='\0';i++){
        ans[text[i]-'a']++;
    }
    int numb = ans['b'-'a'];
    int numa = ans['a'-'a'];
    int numl = ans['l'-'a'];
    int numo = ans['o'-'a'];
    int numn = ans['n'-'a'];

    numl/=2;
    numo/=2;

    int min = numb;
    if(min>numa) min=numa;
    if(min>numl) min=numl;
    if(min>numo) min=numo;
    if(min>numn) min=numn;
    return min;
}

File: week06/20/Pr5.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        ans = 0
        dic = {}
        for x in text :
            if x not in dic:
                dic[x] = 1
            else : dic[x]+=1
        num_b = dic['b'] if 'b' in dic else 0
        num_a = dic['a'] if 'a' in dic else 0
        num_l = dic['l'] if 'l' in dic else 0
        num_o = dic['o'] if 'o' in dic else 0
        num_n = dic['n'] if 'n' in dic else 0

        return min(num_b,num_a,num_l//2,num_o//2,num_n)